src/checks/#cc4a#modern_language.clas.abap

```diff
@@ -0,0 +1,1157 @@
+              REPLACE FIRST OCCURRENCE OF 'SY-TABIX IS NOT INITIAL' IN code_line_index WITH |line_index( { table }[ { key } ] ) <> 0| ##NO_TEXT.
+              REPLACE FIRST OCCURRENCE OF 'SY-TABIX' IN code_line_index WITH |line_index( { table }[ { key } ] )| ##NO_TEXT.
+              code_line_index = |{ code_line_index }.|.
+              APPEND code_line_index TO code_lines.
+            ENDIF.
+        ENDCASE.
+      ELSEIF if_sysubrc <> 0 AND tabix > statement_index + 1.
+        IF <statement>-keyword = 'CALL' OR <statement>-keyword = '+CALL_METHOD'.
+          quickfixable = abap_false.
+          EXIT.
+        ENDIF.
+        CASE if_sysubrc.
+          WHEN 1.
+            token_idx = analyzer->find_clause_index( tokens = <statement>-tokens clause = '=' ).
+            IF token_idx = 0 OR <statement>-tokens[ token_idx + 1 ]-lexeme <> 'ABAP_TRUE'.
+              quickfixable = abap_false.
+              EXIT.
+            ENDIF.
+          WHEN 2.
+            token_idx = analyzer->find_clause_index( tokens = <statement>-tokens clause = '=' ).
+            IF token_idx = 0 OR <statement>-tokens[ token_idx + 1 ]-lexeme <> 'ABAP_FALSE'
+            OR procedure-statements[ tabix + 1 ]-keyword <> 'ENDIF'.
+              quickfixable = abap_false.
+              EXIT.
+            ENDIF.
+        ENDCASE.
+        append_tokens( EXPORTING tokens = <statement>-tokens to_idx = token_idx - 1 CHANGING result = code_line_exists ).
+        IF if_sysubrc = 1.
+          code_line_exists = |{ code_line_exists } = xsdbool( line_exists( { table }[ { key } ] ) ).| ##NO_TEXT.
+        ELSE.
+          code_line_exists = |{ code_line_exists } = xsdbool( NOT line_exists( { table }[ { key } ] ) ).| ##NO_TEXT.
+        ENDIF.
+        APPEND code_line_exists TO code_lines.
+      ENDIF.
+      IF if_sysubrc = 0.
+        EXIT.
+      ENDIF.
+    ENDLOOP.
+
+    IF quickfixable = abap_true AND ( code_line_index IS NOT INITIAL OR code_line_exists IS NOT INITIAL ).
+      DATA(quickfixes) = assistant_factory->create_quickfixes( ).
+      DATA(quickfix) = quickfixes->create_quickfix( c_qf_line_exists ).
+
+      quickfix->replace(
+          context = assistant_factory->create_quickfix_context(
+             VALUE #( procedure_id = procedure-id statements = VALUE #( from = statement_index to = end_idx ) ) )
+                      code = code_lines ).
+      add_finding(
+      EXPORTING
+         procedure = procedure
+         statement_index = statement_index
+         code = c_code_line_exists
+         pseudo_comment = c_ps_line_exists
+         quickfixes = quickfixes
+      CHANGING findings = findings ).
+    ELSE.
+      add_finding(
+      EXPORTING
+         procedure = procedure
+         statement_index = statement_index
+         code = c_code_line_exists
+         pseudo_comment = c_ps_line_exists
+      CHANGING findings = findings ).
+    ENDIF.
+
+  ENDMETHOD.
```

Comment:
This method had 180 lines, any chance of splitting this up into more readable chunks?